src/components/SiteSettingsMenu.tsx

```diff
@@ -166,16 +166,21 @@ function LanguageSection(): ReactElement {
 
 function ThemeSection(): ReactElement {
   const { t } = useTranslation()
-  const { toggleTheme, userDarkMode } = useContext(ThemeContext)
+  const { themeMode, onChangeMode } = useSettings()
+
+  const handleChangeMode = () => {
+    onChangeMode(themeMode === "dark" ? "light" : "dark")
```

Comment:
Since you reuse this code in a couple places, you could also export a `toggleMode` function that uses `onChangeMode` internally to change between light and dark modes. We could still use `onChangeMode` in the future if we want to support custom themes.

Reply:
I embed this code because we are having `system` variant on the `ThemeMode` , so if we will have `System` setting to let user change mode to system setting, we can give "system" as input of `onChangeMode`.

But if we won't use system setting on the UI, then I could make `toggleMode`.